public/app/features/alerting/unified/hooks/useFilteredRules.ts

```diff
@@ -76,37 +77,64 @@ export const useFilteredRules = (namespaces: CombinedRuleNamespace[], filterStat
   return useMemo(() => filterRules(namespaces, filterState), [namespaces, filterState]);
 };
 
+const ufuzzy = new uFuzzy({
```

Comment:
or just note that they're effectively [Damerau–Levenshtein distance](https://en.wikipedia.org/wiki/Damerau%E2%80%93Levenshtein_distance) allowances for each term/word in the needle. (typo tolerance)

Reply:
Yea I kinda guessed as much (though I'm not familiar with the Damerau variant) :)

Reply:
> I'm not familiar with the Damerau variant

it treats adjacent transpositions as an edit distance of 1 instead of 2 (a removal + insertion). now you're familiar with it :rofl: 

but i'll tell you a secret: uFuzzy just constructs a regex with all the variations :monkey_face: 

https://github.com/leeoniya/uFuzzy/blob/1665a1427388e7c77e442dd3ca882882e7f3949d/src/uFuzzy.js#L225-L253